open-apparel-registry/open-apparel-registry
# Fetch next page of facilities via infinite scroll component

## Overview

Adjust facilities tab so that it will fetch the next page of facilities
on reaching the bottom of the infinite scroll list.

Also:

- remove some now unused utility code & tests
- update facilities list endpoint to return results in alphabetical
order
- update url to retrieve facilities to request pages of 25 results at a
time
- add a loading indicator to display contextually at the temporary bottom of the list when there's another page of items to fetch

Connects #742

## Notes

I'm going to make a new issue for scroll restoration. I had worked out part of a solution using a custom hook, but it's most likely an enhancement that we can address later if desired.

## Testing Instructions

- serve this branch, then search for some facilities
- verify that the results return in alphabetical order in the sidebar and that a new page is fetched each time you reach the bottom of the list (until no pages remain)
- verify that any filters you've applied remain applied when fetching a new page of results

## Checklist
